lib/ansible/modules/files/assemble.py

```diff
@@ -204,7 +204,7 @@ def main():
     if validate and "%s" not in validate:
         module.fail_json(msg="validate must contain %%s: %s" % validate)
 
-    path = assemble_from_fragments(src, delimiter, compiled_regexp, ignore_hidden)
+    path = assemble_from_fragments(src, delimiter, compiled_regexp, ignore_hidden, getattr(module, 'tmpdir', None))
```

Comment:
Why are we using getattr everywhere instead of just module.tmpdir?  getattr being non-standard makes people paranoid about making changes but AFAICS, there is no reason to make people afraid of touching this code.

Reply:
mostly to allow people to use the module on older ansible core versions, I tend to err on the side of keeping it compatible if i can

Reply:
K.  I think we should use module.tmpdir then.